src/zarr/storage/_fsspec.py

```diff
@@ -281,6 +283,23 @@ async def delete(self, key: str) -> None:
         except self.allowed_exceptions:
             pass
 
+    async def delete_dir(self, prefix: str) -> None:
+        # docstring inherited
+        if not self.supports_deletes:
+            raise NotImplementedError('This method is only available for stores that support deletes.')
+        if not self.supports_listing:
+            raise NotImplementedError('This method is only available for stores that support directory listing.')
+        self._check_writable()
+
+        path_to_delete = _dereference_path(self.path, prefix)
+
+        # this is probably the same condition as `if self.fs.async_impl`
+        if hasattr(self.fs, "_rm") and inspect.iscoroutinefunction(self.fs._rm):
+            with suppress(*self.allowed_exceptions):
+                await self.fs._rm(path_to_delete, recursive=True)
+        else:
+            raise NotImplementedError("The store does not support async deletes")
```

Comment:
What do we think about further simplifying this to:

```suggestion
        with suppress(*self.allowed_exceptions):
            await self.fs._rm(path_to_delete, recursive=True)
```

I'll let @martindurant comment but I would think Fsspec would be responsible for raising a `NotImplementedError` if this isn't supported by a backend.